onnxruntime/core/session/inference_session.cc

```diff
@@ -542,7 +542,42 @@ int InferenceSession::GetCurrentNumRuns() const {
   return current_num_runs_.load();
 }
 
-common::Status InferenceSession::CheckTypes(MLDataType actual, MLDataType expected) {
+static common::Status CheckShapes(const std::string& input_name,
+                                  const TensorShape& input_shape,
+                                  const TensorShapeProto& expected_shape_proto) {
+  auto input_shape_sz = input_shape.NumDimensions();
+  auto expected_shape = utils::GetTensorShapeFromTensorShapeProto(expected_shape_proto);
```

Comment:
```
  auto expected_shape = utils::GetTensorShapeFromTensorShapeProto(expected_shape_proto);
```
This can be pre-generated for each session.

Reply:
Good point. I see that this is true for this call as well
    auto expected_type = utils::GetMLDataType(*iter->second);

Reply:
fixed